R/getSubsample.R

```diff
@@ -0,0 +1,201 @@
+#' Subsample a \code{SummarizedExperiment} or \code{TreeSummarizedExperiment} object
+#' 
+#' \code{getSubsample} will randomly subsample counts in SE/TSE to return a SE/TSE in
+#' which each sample has same number of total observations/counts/reads. 
+#'
+#' @details
+#' Although the subsampling approach is highly debated in microbiome research, 
+#' we include the \code{getSubsample} function because in some instances these 
+#' can be useful. Note that the output of \code{getSubsample} is a modified SE/TSE.
+#'
+#' @param x A
+#'   \code{SummarizedExperiment} or \code{TreeSummarizedExperiment} object.
+#'
+#' @param abund_values A single character value for selecting the
+#'   \code{SummarizedExperiment} or \code{TreeSummarizedExperiment} \code{assay} 
+#'   used for random subsampling. Only counts are useful and other transformed 
+#'   data as input will give meaningless output.
+#'   
+#' @param min_size A single integer value equal to the number of counts being 
+#'   simulated this can equal to lowest number of total counts 
+#'   found in a sample or a user specified number. 
+#'   
+#' @param seed A random number seed for reproducibility of sampling. 
+#' 
+#' @param replace Logical Default is \code{TRUE}. The default is with 
+#'   replacement (\code{replace=TRUE}). 
+#'   See \code{\link[phyloseq:rarefy_even_depth]{phyloseq::rarefy_even_depth}}
+#'   for details on implications of this parameter.   
+#' 
+#' @param name A single character value specifying the name of transformed
+#'   abundance table.
+#' 
+#' @param verbose Logical Default is \code{TRUE}. When \code{TRUE} an additional 
+#'   message about the random number used is printed.
+#' 
+#' @param ... additional arguments not used
+#' 
+#' @references
+#' McMurdie PJ, Holmes S. Waste not, want not: why rarefying microbiome data 
+#' is inadmissible. PLoS computational biology. 2014 Apr 3;10(4):e1003531.
+#' 
+#' Gloor GB, Macklaim JM, Pawlowsky-Glahn V & Egozcue JJ (2017)
+#' Microbiome Datasets Are Compositional: And This Is Not Optional.
+#' Frontiers in Microbiology 8: 2224. doi: 10.3389/fmicb.2017.02224
+#' 
+#' Weiss S, Xu ZZ, Peddada S, Amir A, Bittinger K, Gonzalez A, Lozupone C, 
+#' Zaneveld JR, Vázquez-Baeza Y, Birmingham A, Hyde ER. Normalization and 
+#' microbial differential abundance strategies depend upon data characteristics. 
+#' Microbiome. 2017 Dec;5(1):1-8.
+#' 
+#' @return a \code{SummarizedExperiment} or \code{TreeSummarizedExperiment} object 
+#'   with subsampled data that is different from input 
+#'   \code{SummarizedExperiment} or \code{TreeSummarizedExperiment}
+#' 
+#' @author Sudarshan A. Shetty 
+#' 
+#' @name getSubsample
+#'  
+#' @examples
+#' # When samples in TSE are less than specified min_size, they will be removed.
```

Comment:
i suggest to use abbreviation TreeSE systematically?

Reply:
Would `subsampleCounts` be an option? I remember in some previous discussion we had avoided `reads` as it may not be that ideal? `rarifyReads` as alias is good for microbiome researchers.

Reply:
Ok with `subsampleCounts` and `rarifyCounts`